### DISCUSSION TITLE
"Answers may vary" flag

### DISCUSSION THREAD

**Discussion by dlglin** [openwebwork/pg#1030] [Ideas]
Some pg problems have more than one correct answer (e.g. when custom checkers are used), but a MathObject has a single correct answer associated with it that is displayed to users.  Can a flag be added to a MathObject that would trigger a message when the correct answers are displayed to say something like "there are other possible correct answers"?

**Comment by somiaj** [comment]
I like this idea. To expand on this though, most problems can accept multiple correct answers (unless some limited context is being used), such as x + 3 vs (4x + 12)/4, vs (2x^2 + 6x)/(2x), or even just 2/3 vs 10/15. Might want to clarify that this is more than just multiple equivalent forms.

**Comment by dpvc** [comment]
Note that you can set `$mo->{correct_ans}` and `$mo->{corrent_ans_latex_string}` to be anything you want, and these are used for the current answer, so you could put whatever message you want there.

If there are a (small) finite number of correct answers, then `parserOneOf.pl` provides a means of checking for that without the need of a custom checker, and it sets these `correct_ans` properties automatically.  You could look at the code for that for ideas about how one might use these fields in a multiple-answer setting.

A flag like you are suggestion could be used in this area:

https://github.com/openwebwork/pg/blob/9cefdf9aed090b9a48d54bcf7060d8d52dc6d37e/lib/Value/AnswerChecker.pm#L90-L93

to adjust these properties of the answer checker, which are set here

https://github.com/openwebwork/pg/blob/9cefdf9aed090b9a48d54bcf7060d8d52dc6d37e/lib/Value/AnswerChecker.pm#L95-L103
